Server/src/services/db/vehicle.controller.ts

```diff
@@ -1,231 +1,329 @@
-	 */
-	public async getByName(name: string, trackId: number): Promise<Vehicle | null> {
-		return await this.prisma.vehicle.findUnique({
-			where: {
-				name_trackId: {
-					name: name,
-					trackId: trackId
-				}
-			},
-			include: {
-				type: true,
-				track: true
-			}
-		})
-	}
-}
+import { PrismaClient, Prisma, Vehicle, VehicleType } from "@prisma/client"
+
+/**
+ * VehicleController class
+ *
+ * Handles vehicle specific access to the datbase.
+ * This controller handles therefore Vehicles and VehicleTypes.
+ *  @functions for VehicleTypes:
+ *      - saveType()
+ *      - updateType()
+ *      - removeType()
+ *      - getAllTypes()
+ *      - getTypeById()
+ *      - getTypeByName()
+ *  @functions for Vehicles:
+ *      - save()
+ *      - update()
+ *      - remove()
+ *      - getAll()
+ *      - getById()
+ *      - getByName()
+ */
+export default class VehicleController {
+	constructor(private prisma: PrismaClient) {}
+
+	// ========================================================= //
+	// [Vehicle Types]
+
+	/**
+	 * Saves a vehicle type in the database.
+	 *
+	 * The parameter are given via object deconstruction from the model `VehicleType`!
+	 * Currently given parameters are:
+	 * @param name - **unique** Name for the type of vehicle.
+	 * @param icon - unique icon name for visualization
+	 * @param description - optional description for the type.
+	 * @returns VehicleType | null if an error occurs
+	 */
+	public async saveType(args: Prisma.VehicleTypeCreateInput): Promise<VehicleType> {
+		// Due to soft-deletion we need to check if a type already exists in an active state
+		if (args.inactive == false) {
+			if ((await this.getTypeByName(args.name)) == null) {
+				// Vehicle doesn't exists in active state
+				return await this.prisma.vehicleType.create({
+					data: args
+				})
+			} else {
+				// VehicleType already exists in active state
+				// Prismas inhouse error for 'unique constraint failed'
+				throw new Prisma.PrismaClientKnownRequestError("Type already exists in active state.", {
+					code: "P2002",
+					clientVersion: ""
+				})
+			}
+		} else {
+			// Operation tried to save an already dead type
+			// Prismas inhouse error for 'query interpretation error'
+			throw new Prisma.PrismaClientKnownRequestError("Tried to create an inactive type.", {
+				code: "P2016",
+				clientVersion: ""
+			})
+		}
+	}
+
+	/**
+	 * Updates a vehicle type in the database.
+	 *
+	 * @param uid - Indicator which vehicle type should be updated.
+	 *
+	 * The parameter are given via object deconstruction from the model `VehicleType`!
+	 * Currently given parameters are:
+	 * @param name - New name of the vehicle type after change. (Optional)
+	 * @param description - New description of the vehicle type after change. (Optional)
+	 * @returns
+	 */
+	public async updateType(uid: number, args: Prisma.VehicleTypeUpdateInput): Promise<VehicleType> {
+		if (args.inactive == false) {
```

Comment:
```suggestion
		if (args.inactive == true) {
```

Reply:
Actually, it's correct with `false`: We are throwing an error if we are trying to ressurrect an `VehicleType`. Therefore setting `inactive = false` should not be a valid operation for update.